doc/users/next_whats_new/2020-04-09-datetime-epoch-change.rst

```diff
@@ -0,0 +1,22 @@
+Dates now use a modern epoch
+----------------------------
+
+Matplotlib must convert dates to floating point numbers for plotting, which
+is done as floating point days since an epoch.  Previously, `.dates.date2num`
```

Comment:
OK rewrote the first paragraph.  I guess with "convert" I'm trying to make the machinery a little less mysterious and give the users some idea of what is going on behind the scenes:

```
Matplotlib converts dates to days since an epoch using `.dates.date2num` (via
`matplotlib.units`).  Previously, an epoch of ``0000-12-31T00:00:00`` was used
so that ``0001-01-01`` was converted to 1.0.  An epoch so distant in the
past meant that a modern date was not able to preserve microseconds because
2000 years times the 2^(-52) resolution of a 64-bit float gives 14
microseconds.
```

Reply:
For me personally, "convert" triggers "why are they changing the dates I provided".